src/WebJobs.Script.WebHost/Middleware/FunctionInvocationMiddleware.cs

```diff
@@ -158,6 +156,34 @@ private async Task<IActionResult> GetResultAsync(HttpContext context, IFunctionE
             return new OkResult();
         }
 
+        private void PopulateRouteData(HttpContext context)
+        {
+            // Add route data to request info
+            // TODO: Keeping this here for now as other code depend on this property, but this can be done in the HTTP binding.
+            var routingFeature = context.Features.Get<IRoutingFeature>();
+
+            var routeData = new Dictionary<string, object>(routingFeature.RouteData.Values);
+
+            // Get optional parameters that were not used and had no default
+            Route functionRoute = routingFeature.RouteData.Routers.FirstOrDefault(r => r is Route) as Route;
+
+            if (functionRoute != null)
+            {
+                var optionalParameters = functionRoute.ParsedTemplate.Parameters.Where(p => p.IsOptional && p.DefaultValue == null);
+
+                foreach (var parameter in optionalParameters)
+                {
+                    // Make sure we didn't have the parameter in the values dictionary
+                    if (!routeData.ContainsKey(parameter.Name))
+                    {
+                        routeData.Add(parameter.Name, null);
```

Comment:
With this change, we'll have binding data for the parameter, but not set, which will bind to the default parameter value.

Reply:
huh?